### PR TITLE
Carer first and last name

### PR DESCRIPTION
**Pull request by katbow**
for #569 #600

Updates carer first and last name when connecting as a carer.

Does not allow the carer to connect as an SU after this

Does not allow the carer to update their name

### CODE REVIEW

File: lib/healthlocker/slam/carer_connection.ex [hunk 1]
```diff
@@ -5,18 +5,18 @@ defmodule Healthlocker.Slam.CarerConnection do
   alias Healthlocker.{EPJSUser, ReadOnlyRepo}
 
   embedded_schema do
-    field :first_name
-    field :last_name
+    field :forename
```
Comment (nelsonic) [inline lib/healthlocker/slam/carer_connection.ex]: can't say I'm a "_fan_" of `forename` over `first_name` but if that's what the PO wants... 👍

File: priv/repo/migrations/20170510095252_add_first_last_name_to_users.exs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+defmodule Healthlocker.Repo.Migrations.AddFirstLastNameToUsers do
+  use Ecto.Migration
+
+  def change do
+    alter table(:users) do
+      add :first_name, :string
```
Comment (nelsonic) [inline priv/repo/migrations/20170510095252_add_first_last_name_to_users.exs]: should these fields be `forename` and `surname` ?

File: test/controllers/slam_controller_test.exs [hunk 3]
```diff
@@ -1,6 +1,30 @@
 defmodule Healthlocker.SlamControllerTest do
   use Healthlocker.ConnCase
-  alias Healthlocker.User
+  alias Healthlocker.{User, EPJSUser, ReadOnlyRepo}
+
+  @valid_attrs %{
+    first_name: "Kat",
```
Comment (nelsonic) [inline test/controllers/slam_controller_test.exs]: so the person can have 4 (_four_) names?
+ `first_name`
+ `last_name`
+ `forename`
+ `surname`

or is this just while we are _migrating_ the data?

File: test/lib/slam/carer_connection_test.exs [hunk 4]
```diff
@@ -4,14 +4,14 @@ defmodule Healthlocker.Slam.CarerConnectionTest do
   alias Healthlocker.{EPJSUser, ReadOnlyRepo}
 
   @valid_attrs %{
-    first_name: "Kat",
-    last_name: "Bow",
+    forename: "Kat",
```
Comment (nelsonic) [inline test/lib/slam/carer_connection_test.exs]: is `forename` only used for carers?

**Comment by katbow** [comment]
@nelsonic forename/surname is for the ePJS check. I changed the previous first_name/last_name to forename/surname because that is the data used to check the patient health record, and to differentiate from the new `first_name` & `last_name` we are using in HL. 

So if a user enters details to connect as a carer, they now enter their own `first_name` and `last_name` (HL), then all of the details of the person they are caring for which includes `forename`, `surname`, `date_of_birth` and `nhs_number` (ePJS).

**Comment by katbow** [comment]
all done @tonydaly

**Pull request merged**